### PR TITLE
Added list component

### LINKED ISSUE #45: List Component
I'm going to write a list component that allows you to have 1 or 2 lines of text and an icon or avatar on one or both the left and right side. 

This component will have an outside component and an child component. The outside component will house all the child component and will be in charge of layout. 

The child component will be in charge of laying out the list-items content: text, icon and/or avator

I'm thinking it'll it'll look something like this:

```html
<hc-list>
     <hc-list-item>
          <div hc-avatar></div>
          Main text
          <div hc-subtext>I'm subtext</div>
          <hc-icon class="fa-filter"></hc-icon>
     </hc-list-item>
</hc-list>
```

I'd also like to use this for the side navigation, which I'm also working #42

### PR DESCRIPTION
**Pull request by jtrujill**
Closes #45

### CODE REVIEW

File: src/app/lib/list/list-demo/list-demo.component.html [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        &lt;hc-icon fontSet=&quot;fa&quot; fontIcon=&quot;fa-plus&quot;&gt;&lt;/hc-icon&gt;
+      &lt;/hc-list-item&gt;
+    &lt;/hc-list&gt;
+  &lt;/div&gt;
+          </code>
+      </pre>
+  <br>
+  <hr style="height:1px;border:none;background-color:#E5E5E5;">
+  <br>
+  <div class="headerName">Components and Directives</div>
```
Comment (benjanderson) [inline src/app/lib/list/list-demo/list-demo.component.html]: I would love to see a demonstration of all the components in this demo. It seems like the following were left out. Even after reviewing the code, I am not sure what they are for

- hc-list-icon
- hc-avatar
- hc-line

File: src/app/lib/list/list-demo/list-demo.component.html [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    <p>This is the container for the hc-list-items</p>
+  </div>
+
+  <div class="directive-description">
+    <h4>hc-list-item</h4>
+    <p>An item within the hc-list</p>
+  </div>
+
+  <div class="directive-description">
+    <h4>hcListIcon</h4>
```
Comment (benjanderson) [inline src/app/lib/list/list-demo/list-demo.component.html]: some of these have hyphens and some dont. hcListIcon is the first. I would like this to be a little more consistent

image

Comment (jtrujill) [reply src/app/lib/list/list-demo/list-demo.component.html]: hc-list and hc-list-item are components. hcListIcon, hcAvatar and hcLine are directives. I can clarify in the description what they are

**Comment by jtrujill** [comment]
I've updated the demo page so it's styles are consistent with the rest of the demo pages

File: src/app/lib/list/list-demo/list-demo.component.html (hunk 1 above)
Comment (jtrujill) [reply src/app/lib/list/list-demo/list-demo.component.html]: hcListAvatar, hcListLine and hcListIcon are used for content projection. They are basically markers. 

hcListAvatar will position the element on the left side of the list-item with a border-radius of 50%. 

hcListIcon will position it to the left. 

hcListLine marks the element as your main content and will position it in the center

I'm also able to give these elements special styling; however, the styling is pretty light in this early implementation

**Comment by jtrujill** [comment]
@benjanderson I believe I have those changes merged in that you requested

**Comment by benjanderson** [comment]
@jtrujill did you push up your changes? I dont see them

**Comment by jtrujill** [comment]
https://github.com/HealthCatalyst/Fabric.Cashmere/pull/69#issuecomment-362827488
Yes I amended my commit and pushed the changes. It updates the demo page to have the same styling as the rest of the site and I've fixed some of the documentation. Maybe in the future I'll squash my commits after the review, so the reviewers can see my changes

**Pull request merged**